### PR TITLE
Limit de $q_orders results

### PR DESCRIPTION
**Pull request by djbuch**
We limit this query to avoid having an IN query with thousands of unnecessary order ids.

I have multiplied the limit by 4 to minimize the case where products have been disabled.

But having hundreds of IDs in the order ID list or thousands or even hundred of thousands ID in this list (for big shops) make the whole difference between a query that runs a few ms and a query that runs for seconds.

Fixes PrestaShop/PrestaShop#20874

### CODE REVIEW

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Is there a reason to use the ORDER BY RAND()?
Could you create an issue on the main repository to be sure the Product team agree about this improvement? :)

**Review by matthieu-rolland — COMMENTED** [review]
I agree with @PierreRambaud , you should probably create an issue here, explaining your problem, and link this issue to your PR.

Here's where to create your issue: https://github.com/PrestaShop/PrestaShop/issues/new?template=1_bug_report.md

Then it will be validated by Prestashop's product team 👍

**Comment by djbuch** [comment]
Hello I created this issue here : https://github.com/PrestaShop/PrestaShop/issues/20874

I'm adding ORDER BY RAND() because of the ORDER BY RAND() present in the query for the products line 306.

If I don't add the ORDER BY RAND() in the first query it will always select the same orders to find cross sellings from.

File: ps_crossselling.php [hunk 1]
```diff
@@ -247,7 +247,7 @@ protected function getOrderProducts(array $productIds = array())
         FROM '._DB_PREFIX_.'orders o
         LEFT JOIN '._DB_PREFIX_.'order_detail od ON (od.id_order = o.id_order)
         WHERE o.valid = 1
-        AND od.product_id IN ('.implode(',', $productIds).')';
+        AND od.product_id IN ('.implode(',', $productIds).') ORDER BY RAND() LIMIT '.(int)Configuration::get('CROSSSELLING_NBR')*4;
```
Comment (PierreRambaud) [inline ps_crossselling.php:250]:
```suggestion
        AND od.product_id IN ('.implode(',', $productIds).') ORDER BY RAND() LIMIT ' . (int) Configuration::get('CROSSSELLING_NBR')*4;
```

**Review by PierreRambaud — APPROVED** [review]

**Comment by djbuch** [comment]
Here you have also the pull request for the core : https://github.com/PrestaShop/PrestaShop/pull/20916

**Review by matthieu-rolland — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @djbuch,

Please resolve the Conflicting files.

Thanks!

**Comment by sarahdib** [comment]
Hello

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to ask to re-open when it will be possible to fix the conflicting file

Thank you!

**Pull request closed without merging**